File: Frontend/src/services/ContractService.js
import axios from "axios";
import React from 'react'

const CONTRACT_API_BASE_URL =`http://localhost:8080/contracts/`

class ContractService{
    addContract(contract){
        console.log('ana f service')
        return axios.post(CONTRACT_API_BASE_URL + "contract" , contract)
    }


    getContractsByEquipement(numSerie){
        return axios.get(CONTRACT_API_BASE_URL +  numSerie)
    }

    getContractByMarche(marche){
        return axios.get(CONTRACT_API_BASE_URL + "contratByMarche/" + marche)
    }


    updateContract(contract){
        return axios.post(CONTRACT_API_BASE_URL + "editContract", contract)
    }

    deleteContract(marche){
        return axios.post(CONTRACT_API_BASE_URL + "deleteContract/" + marche)
    }
    getContractCount(){
        return axios.get(CONTRACT_API_BASE_URL + "getContractCount")
    }
    }

export default new ContractService();

File: Backend/src/main/java/com/example/demo/service/ContractService.java
package com.example.demo.service;

import com.example.demo.model.Contract;
import com.example.demo.model.Contrat;
import com.example.demo.model.IContractByEquipement;
import com.example.demo.model.ICounter;
import com.example.demo.repository.ContractRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ContractService {

    @Autowired
    private ContractRepository contractRepository;



    public void saveContract(Contrat contract){
        contractRepository.addContract(contract.getMarche(),contract.getEquipement_num_serie(),contract.getDate_intervention(),contract.getPrestataire(),contract.getIntervenant(),contract.getAction(),contract.getCommentaire());
    }

    public List<IContractByEquipement> getContractsEquipement(String equipement_num_serie){
        return contractRepository.getContractsByEquipement(equipement_num_serie);
    }
    public IContractByEquipement getContractMarche(String marche){
        return contractRepository.getContractByMarche(marche);
    }

    public void editContract(Contrat contract){
        contractRepository.editContract(contract.getMarche(),contract.getEquipement_num_serie(),contract.getDate_intervention(),contract.getPrestataire(),contract.getIntervenant(),contract.getAction(),contract.getCommentaire());
    }

    public void deleteContract(String marche){
         contractRepository.deleteContract(marche);
    }

    public ICounter getContactCount(){
        return contractRepository.getContractCount();
    }
}
